### ISSUE TITLE
Reversed roles with CHARSET negotiation

### ISSUE THREAD

**Issue by amotzkau** [jquast/telnetlib3#59]
It seems that telnetlib3 sends the CHARSET subnegotiation upon receiving a WILL CHARSET (and waits for the subnegotiation after receiving a DO CHARSET).

RFC 2066 says about the IAC SB CHARSET REQUEST:
> This message initiates a new CHARSET subnegotiation.  It can only be sent by a side that has received a DO CHARSET message and sent a WILL CHARSET message (in either order).

This is exactly the other way around.

And it also seems that telnetlib3 doesn't respond to a WILL CHARSET with a DO CHARSET, which I think it should.

**Comment by jquast** [comment]
@amotzkau can you suggest any telnet client and server that supports CHARSET negotiation?

I vaguely remember implementing against "Atlantis" MUD telnet client, which was the only reference I could find compatible with my computers at the time I developed it.

**Comment by amotzkau** [comment]
Mud clients (that I know of):
* TinyFugue Rebirth: https://github.com/ingwarsw/tinyfugue
* Mudlet: https://github.com/Mudlet/Mudlet

Except the one that we have, I'm not aware of servers implementing that:
* UNItopia (unitopia.de, Port 23): https://www.unitopia.de/pub/UNItopia/mudlib.tar.gz

**Comment by gtaylor** [comment]
Would this bug be causing us to hit `connect_maxwait` like this?
```
2023-02-19 00:43:58,216 DEBUG server.py:155 encoding failed after 4.00s: US-ASCII
```

**Comment by jquast** [comment]
I might consider just removing charset temporarily until a correct implementation can tested.

**Comment by gtaylor** [comment]
That may be safest. Probably not of much consolation, but Mudlet somehow ends up in the correct state despite the non-compliant negotiation!

**Comment by jquast** [comment]
Yeah I see that in the logs I've got mudlet installed and I hope to work it out over the weekend for next release

**Comment by gtaylor** [comment]
Cool! I was trying to take a shot at this but my unfamiliarity with the codebase made it impossible in the time I had. Happy to help test and refine once you've got the basics in place!

**Comment by jquast** [comment]
2.0.7 has minor bugfix: "respond WILL CHARSET with DO CHARSET" which causes unitopia.de to negotiate (utf8) charset successfully with telnetlib3-client, thank you all!
